src-self-hosted/ir.zig

```diff
@@ -350,16 +344,10 @@ pub const Inst = struct {
             return self.args.len + 1;
         }
         pub fn getOperand(self: *const Call, index: usize) ?*Inst {
-            var i = index;
-
-            if (i < 1)
+            if (index == 0)
                 return self.func;
-            i -= 1;
-
-            if (i < self.args.len)
-                return self.args[i];
-            i -= self.args.len;
-
+            if (index - 1 < self.args.len)
+                return self.args[index - 1];
```

Comment:
The previous pattern here is defensive coding against bugs as @Vexu notes. If you look elsewhere in  the file, this is the One Way To Do Things that is used for iterating nodes. This new code is more bug prone if things are added or removed from the struct.